File: algorithms/0009.palindrome_number/palindrome_number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let s = x.to_string();
        for (c, n) in s.as_bytes().iter().zip(s.as_bytes().iter().rev()) {
            if c != n {
                return false;
            }
        }
        return true;
    }
}

File: algorithms/0009.palindrome_number/palindrome_number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int64_t reserve_n = 0;
        int64_t n = x;
        while (n > 0) {
            reserve_n *= 10;
            reserve_n += (n % 10);
            n /= 10;
        }
        return reserve_n == x;
    }
};